src/Common/src/System/Net/Http/TlsCertificateExtensions.cs

```diff
@@ -73,6 +73,24 @@ private static bool IsClientCertificate(X509Certificate2 cert)
             return isClientAuth && isDigitalSignature;
         }
 
+        internal static X509Chain BuildNewChain(X509Certificate2 certificate)
+        {
+            var chain = new X509Chain();
+            chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllFlags;
+            chain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
+            chain.ChainPolicy.ApplicationPolicy.Add(s_clientCertOidInst);
```

Comment:
In Automatic mode it has to match the presented allowed CAs list and be a valid client auth cert (so the chain is part of the filter).  In manual mode where you've given a single cert there's no filtering, just doing.

Since the single manual cert is being chosen no matter what this method does, it seems better to let the chain build so that intermediate CA certs are sent.